### ISSUE TITLE
Should we allow arithmetic between STmag and ABmag?

### ISSUE THREAD

**Issue by pllim** [astropy/astropy#13248]
As a follow-up of https://github.com/astropy/astropy/pull/13158#discussion_r858962549 and spacetelescope/synphot_refactor#327 , should this be allowed? It is currently possible (at the time of posting).

```python
>>> import astropy
>>> astropy.__version__
'5.2.dev64+g6ed769d58'
>>> from astropy import units as u
>>> (5 * u.STmag - 2.5 * u.ABmag)
<Magnitude 2.5 mag(ST / AB)>
```

Might be related: #12472

cc @mhvk @olebole

**Comment by mhvk** [comment]
I think this should just be allowed: the magnitudes really are just log of whatever is inside, so what it does is internally consistent.
We could put this as in the docs an example of something that may at first be surprising, though.

**Comment by aaryapatil** [comment]
@mhvk Would it be okay if I work through this issue with the students at the Pan-African School for Emerging Astronomers and open a PR (similar to how you ran your open-source/Astropy workshop at Toronto)? It is a documentation issue which is easier to solve for first time contributors. Since the units information is a bit more involved, I can help the students with specific information to add. If you would prefer someone else do it, that is fine too!

Just to clarify, we should add the above as an example of a "surprising" case in https://docs.astropy.org/en/stable/units/logarithmic_units.html#arithmetic-and-photometric-applications, right?

**Comment by mhvk** [comment]
Yes, that location sounds good.  I think that one would also want to add a "solution" of how to compare measurements in two magnitude systems.  In this case, maybe one can discuss via examples like
```
m1 = 5*u.STmag
m1 = 5*u.STmag
m1 - m2
# <Magnitude 0. mag(ST / AB)>
m1 - m2.to(u.STmag, equivalencies=u.spectral_density(5500.*u.AA))
# <Magnitude -0.00976169 mag>
# But
m1 - m2.to(u.STmag, equivalencies=u.spectral_density(4400.*u.AA))
# <Magnitude 0.47478837 mag>
```
(This will probably help people to understand how those ST and AM magnitudes work too...)

p.s.  Alternatively, this could become a small tutorial I guess

**Comment by pllim** [comment]
@aaryapatil , do you want to scar those poor students with magnitudes so early in their careers? 😆

**Comment by aaryapatil** [comment]
Haha, I already did 😉
